meta-lxatac-software: panel-mipi-dbi: cleanups for upstreaming

hnez opened the pull request:
We want to upstream the `panel-mipi-dbi` class and `-native` recipes (as part of internal task number 050.441).

I think [openembedded/meta-openembedded](https://github.com/openembedded/meta-openembedded) would be a good fit for such a upstreaming effort.

These are some cleanups to the recipes before making the first upstreaming attempt.

Feel free to be extra strict now, so no one is going to be mean to me on the mailing list.

hnez committed:
meta-lxatac-software: panel-shineworld-lh133k: update to match bbclass doc

The `panel-mipi-dbi.bbclass` documentation has the `inherit` at the top of
the file so we should do so as well.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: panel-mipi-dbi-native: rename to ..._git.bb

The upstream project did not have a release yet (the SRCREV we use is
indeed only the second commit in the repository).
Indicate that with a `_git` version suffix.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: panel-mipi-dbi-native: update to current HEAD

The `python3` shebang has been mainlined, so there is no need to carry
it locally anymore. This is the only change in this update.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: panel-mipi-dbi-native: improve package metadata

Update `SUMMARY`, `DESCRIPTION`, `HOMEPAGE` and `SECTION` in preparation
to mainlining of the recipe to meta-openembedded.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-software: classes: panel-mipi-dbi: add usage documentation

Make the class more usable to the general public in preparation to
mainlining to meta-openembedded.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

ejoerns reviewed (approved):
Changes look fine to me.

Two additional note on the class:

For meta-oe `master` I guess `PANEL_FIRMWARE` needs to be looked up in `UNPACKDIR` rather than in `WORKDIR`.

I'd also suggest placing the output of `do_compile()` into `B` instead of `WORKDIR`.

hnez committed:
meta-lxatac-software: classes: panel-mipi-dbi: compile to B not WORKDIR

According to the Variable glossary[1] `B` is:

  The directory within the Build Directory in which the OpenEmbedded build
  system places generated objects during a recipe’s build process.

[1]: https://docs.yoctoproject.org/ref-manual/variables.html#term-B

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
> For meta-oe `master` I guess `PANEL_FIRMWARE` needs to be looked up in `UNPACKDIR` rather than in `WORKDIR`.

That makes sense, thanks.
I've updated my styhead development branch to include this branch and the `UNPACKDIR` update:

https://github.com/linux-automation/meta-lxatac/blob/7e1b5ba2ac941f8dee0ee449dfe4e10d3a086e67/meta-lxatac-software/classes/panel-mipi-dbi.bbclass#L36-L40

> I'd also suggest placing the output of `do_compile()` into `B` instead of `WORKDIR`.

Oh, yes, thanks! Updated as well.
